lib/errors/error-collector.js

```diff
@@ -89,69 +89,47 @@ class ErrorCollector {
   }
 
   /**
-   * Helper method for processing errors that are created with .noticeError()
+   * Gets the iterable property from the transaction based on the error type
+   * @param {Transaction} transaction the collected exception's transaction
+   * @param {string} errorType the type of error to be processed: "user", "transactionException", "transaction"
+   * @returns
+   */
+  _getIterableProperty(transaction, errorType) {
+    let iterableProperty = null
+    if (errorType === 'user' && transaction.userErrors.length) {
+      iterableProperty = transaction.userErrors
+    }
+    if (errorType === 'transactionException') {
+      iterableProperty = transaction.exceptions
+    }
+    return iterableProperty
+  }
+
+  /**
+   * Helper method for processing errors that are created with .noticeError(), exceptions
+   * on the transaction (transaction.exceptions array), and inferred errors based on Transaction metadata.
    *
    * @param {Transaction} transaction the collected exception's transaction
    * @param {number} collectedErrors the number of errors we've successfully .collect()-ed
    * @param {number} expectedErrors the number of errors marked as expected in noticeError
-   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post-processing
+   * @param {string} errorType the type of error to be processed; "user", "transactionException", "transaction"
+   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post processing
    */
-  _processUserErrors(transaction, collectedErrors, expectedErrors) {
-    if (transaction.userErrors.length) {
-      for (let i = 0; i < transaction.userErrors.length; i++) {
-        const exception = transaction.userErrors[i]
+  _processErrors(transaction, collectedErrors, expectedErrors, errorType) {
+    const iterableProperty = this._getIterableProperty(transaction, errorType)
+    if (iterableProperty) {
+      iterableProperty.forEach((exception) => {
         if (this.collect(transaction, exception)) {
           ++collectedErrors
-          // if we could collect it, then check if expected
           if (
             urltils.isExpectedError(this.config, transaction.statusCode) ||
             errorHelper.isExpectedException(transaction, exception, this.config, urltils)
           ) {
             ++expectedErrors
           }
         }
-      }
-    }
-
-    return [collectedErrors, expectedErrors]
-  }
```

Comment:
I find it helpful if the main logic of a function is at the leftmost indentation possible. Consider the following functions:

```js
function foo (bar) {
  let result
	if (bar != null) { // coercive check to cover null and undefined
		result = ''
		for (const ele of bar) {
			result += ele.baz
		}
	}

	return result
```

vs

```js
function foo (bar) {
	if (bar == null) { // coercive check to cover null and undefined
		return
	}

	let result = ''
	for (const ele of bar) {
		result += ele.baz
	}

	return result
}
```

In my opinion, it's easier in the second version to see and review the purpose of the function. Versus the first version where we have to reason through nesting to find the main purpose of the function.